src/IFeeCurrency.sol

```diff
@@ -30,41 +30,37 @@ import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-    /// that can be used by the transaction.
-    /// - The implementation must deduct `value` from `from`'s balance.
-    /// - Must revert if `msg.sender` is not the zero address.
+    // @notice Called before transaction execution to reserve the maximum amount of gas
+    // that can be used by the transaction.
+    // - The implementation must deduct `value` from `from`'s balance.
+    // - Must revert if `msg.sender` is not the zero address.
     function debitGasFees(address from, uint256 value) external;
 
-    /// @notice New function signature, will be used when all fee currencies have migrated.
-    /// Credited amounts include gas refund, base fee and tip. Future additions
-    /// may include L1 gas fee when Celo becomes and L2.
-    /// - The implementation must increase each `recipient`'s balance by corresponding `amount`.
-    /// - Must revert if `msg.sender` is not the zero address.
-    /// - Must revert if `recipients` and `amounts` have different lengths.
+    // @notice New function signature, will be used when all fee currencies have migrated.
+    // Credited amounts include gas refund, base fee and tip. Future additions
+    // may include L1 gas fee when Celo becomes and L2.
+    // - The implementation must increase each `recipient`'s balance by corresponding `amount`.
+    // - Must revert if `msg.sender` is not the zero address.
+    // - Must revert if `recipients` and `amounts` have different lengths.
     function creditGasFees(address[] calldata recipients, uint256[] calldata amounts) external;
 
-    /// @notice Old function signature for backwards compatibility
-    /// - Must revert if `msg.sender` is not the zero address.
-    /// - `refund` must be credited to `from`
-    /// - `tipTxFee` must be credited to `feeRecipient`
-    /// - `baseTxFee` must be credited to `communityFund`
-    /// - `gatewayFeeRecipient` and `gatewayFee` only exist for backwards
-    ///   compatibility reasons and will always be zero.
+    // @notice Old function signature for backwards compatibility
+    // - Must revert if `msg.sender` is not the zero address.
+    // - `refund` must be credited to `from`
+    // - `tipTxFee` must be credited to `feeRecipient`
+    // - `baseTxFee` must be credited to `communityFund`
+    // - `gatewayFeeRecipient` and `gatewayFee` only exist for backwards
+    //   compatibility reasons and will always be zero.
     function creditGasFees(
-        address from,
-        address feeRecipient,
-        address gatewayFeeRecipient,
-        address communityFund,
-        uint256 refund,
-        uint256 tipTxFee,
-        uint256 gatewayFee,
-        uint256 baseTxFee
+        address refundRecipient,
+        address tipRecipient,
+        address _gatewayFeeRecipient,
+        address baseFeeRecipient,
+        uint256 refundAmount,
+        uint256 tipAmount,
+        uint256 _gatewayFeeAmount,
+        uint256 baseFeeAmount
```

Comment:
I intentionally preserved the old names, since they are part of the ABI. I agree the new names are better, but is it really safe to change them?

If we change the names, the comment should match the new names.